### ISSUE TITLE
Applying `TLSRoute` example doesn't attach the route to Gateway

### ISSUE THREAD

**Issue by pmalek** [Kong/kubernetes-ingress-controller#3575]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When applying the [examples/gateway-tlsroute.yaml](https://github.com/Kong/kubernetes-ingress-controller/blob/main/examples/gateway-tlsroute.yaml), the route does not get attache to the Gateway.

Controller logs the following:

```
[ingress-controller] time="2023-02-17T09:59:47Z" level=info msg="processing tlsroute" GatewayV1Alpha2TLSRoute="{\"Namespace\":\"kong\",\"Name\":\"tlsecho\"}" logger=controllers.TLSRoute name=tlsecho namespace=kong
[ingress-controller] time="2023-02-17T09:59:47Z" level=info msg="checking deletion timestamp" GatewayV1Alpha2TLSRoute="{\"Namespace\":\"kong\",\"Name\":\"tlsecho\"}" logger=controllers.TLSRoute name=tlsecho namespace=kong
[ingress-controller] time="2023-02-17T09:59:47Z" level=info msg="retrieving GatewayClass and Gateway for route" GatewayV1Alpha2TLSRoute="{\"Namespace\":\"kong\",\"Name\":\"tlsecho\"}" logger=controllers.TLSRoute name=tlsecho namespace=kong
[ingress-controller] time="2023-02-17T09:59:47Z" level=info msg="checking if the tlsroute's gateways are ready" GatewayV1Alpha2TLSRoute="{\"Namespace\":\"kong\",\"Name\":\"tlsecho\"}" logger=controllers.TLSRoute name=tlsecho namespace=kong
[ingress-controller] time="2023-02-17T09:59:47Z" level=info msg="ensuring status contains Gateway associations" GatewayV1Alpha2TLSRoute="{\"Namespace\":\"kong\",\"Name\":\"tlsecho\"}" logger=controllers.TLSRoute name=tlsecho namespace=kong
[ingress-controller] time="2023-02-17T09:59:47Z" level=info msg="tlsroute not configured,requeueing" GatewayV1Alpha2TLSRoute="{\"Namespace\":\"kong\",\"Name\":\"tlsecho\"}" logger=controllers.TLSRoute name=tlsecho namespace=kong
```

Whereas the Gateway has the following status:

```
status:
  addresses:
  - type: IPAddress
    value: 172.18.1.1
  conditions:
  - lastTransitionTime: "2023-02-17T09:59:44Z"
    message: this unmanaged gateway has been picked up by the controller and will
      be processed
    observedGeneration: 2
    reason: Accepted
    status: "True"
    type: Accepted
  - lastTransitionTime: "2023-02-17T09:59:44Z"
    message: addresses and listeners for the Gateway resource were successfully updated
    observedGeneration: 2
    reason: Ready
    status: "True"
    type: Ready
  listeners:
  - attachedRoutes: 0
    conditions:
    - lastTransitionTime: "2023-02-17T09:59:44Z"
      message: ""
      observedGeneration: 2
      reason: NoConflicts
      status: "False"
      type: Conflicted
    - lastTransitionTime: "2023-02-17T09:59:44Z"
      message: the listener is ready and available for routing
      observedGeneration: 2
      reason: Ready
      status: "True"
      type: Ready
    - lastTransitionTime: "2023-02-17T09:59:44Z"
      message: ""
      observedGeneration: 2
      reason: ResolvedRefs
      status: "True"
      type: ResolvedRefs
    name: http
    supportedKinds:
    - group: gateway.networking.k8s.io
      kind: HTTPRoute
  - attachedRoutes: 0
    conditions:
    - lastTransitionTime: "2023-02-17T09:59:44Z"
      message: no Kong listen with the requested protocol is configured
      observedGeneration: 2
      reason: UnsupportedProtocol
      status: "True"
      type: Accepted
    - lastTransitionTime: "2023-02-17T09:59:44Z"
      message: the listener is not ready and cannot route requests
      observedGeneration: 2
      reason: Pending
      status: "False"
      type: Ready
    name: tls
    supportedKinds:
    - group: gateway.networking.k8s.io
      kind: TLSRoute
```

### Expected Behavior

Route attaches successfully and traffic can get through.

### Steps To Reproduce

```markdown
1. Run KIC with Gateway.
2. Apply https://github.com/Kong/kubernetes-ingress-controller/blob/main/examples/gateway-tlsroute.yaml
3. Observe `TLSRoute` not routing traffic and not being attached to Gateway.
```

### Kong Ingress Controller version

fbc3cbaee8337902295d5e023c271444a5b3e813

### Kubernetes version

```shell
v1.26.0
```

**Comment by mlavacca** [comment]
The TLSRoute status looks wrong: the `accepted` condition is true, but the reason is `UnsupportedProtocol `. With such a reason, the `accepted` condition should be false.

**Comment by mlavacca** [comment]
fixed by #3579
